ams/ResetCommand.cs

```diff
@@ -0,0 +1,85 @@
+﻿using AMSMigrate.Ams;
+using AMSMigrate.Contracts;
+using Azure;
+using Azure.Core;
+using Azure.ResourceManager.Media;
+using Azure.Storage.Blobs;
+using Azure.Storage.Blobs.Models;
+using Microsoft.Extensions.Logging;
+using Spectre.Console;
+
+namespace AMSMigrate.ams
+{
+    internal class ResetCommand : BaseMigrator
+    {
+        private readonly ILogger _logger;
+        private readonly ResetOptions _options;
+        private readonly IMigrationTracker<BlobContainerClient, AssetMigrationResult> _tracker;
+        public ResetCommand(GlobalOptions globalOptions,
+            ResetOptions resetOptions,
+            IAnsiConsole console,
+            TokenCredential credential,
+            IMigrationTracker<BlobContainerClient, AssetMigrationResult> tracker,
+            ILogger<ResetCommand> logger)
+         : base(globalOptions, console, credential)
+        {
+            _options = resetOptions;
+            _logger = logger;
+            _tracker = tracker;
+        }
+
+        public override async Task MigrateAsync(CancellationToken cancellationToken)
+        {
+            var account = await GetMediaAccountAsync(_options.AccountName, cancellationToken);
+            _logger.LogInformation("Begin reset assets on account: {name}", account.Data.Name);
+
+            var storage = await _resourceProvider.GetStorageAccountAsync(account, cancellationToken);       
```

Comment:
No need to pre-set a primary storage account before getting a list of assets.  individual asset might be in non-primary storage account.  It is better to figure out the actual storage account and container inside the for loop in line 42.